packages/data-stores/src/site/types.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+
+import { Action } from 'redux';
+
+export const enum ActionType {
+	CREATE_SITE = 'CREATE_SITE',
+	RECEIVE_NEW_SITE = 'RECEIVE_NEW_SITE',
+	RECEIVE_NEW_SITE_SUCCESS = 'RECEIVE_NEW_SITE',
+	RECEIVE_NEW_SITE_FAILED = 'RECEIVE_NEW_SITE_FAILED',
+	FETCH_NEW_SITE = 'FETCH_NEW_SITE',
+}
+
+export interface NewSite {
+	blogname: string;
+	blogid: string;
+	blog_details: object;
+}
+
+export interface NewSiteSuccessResponse {
+	success: boolean;
+	blog_details: object;
+}
+
+export interface NewSiteErrorResponse {
```

Comment:
Just a reminder: check the API for the varying error responses. I looked at the callback in the API code and yes, they vary :)